code/test/Fakes/MSBuild/FakeMsBuildProject.cs

```diff
@@ -112,30 +112,32 @@ private bool ItemExists(string itemPath)
 
         private VsItemType GetItemType(string fileName)
         {
-            string ext = Path.GetExtension(fileName).ToLower();
-            if (ext == ".cs")
-            {
-                if (fileName.ToLower().Contains(".xaml.cs"))
-                {
-                    return VsItemType.CompiledWithDependant;
-                }
-                else
-                {
-                    return VsItemType.Compiled;
-                }
-            }
-            else if (ext == ".xaml")
-            {
-                return VsItemType.XamlPage;
-            }
-            else if (ext == ".resw")
-            {
-                return VsItemType.Resource;
-            }
-            else
+            VsItemType returnType = VsItemType.Content;
+
+            switch (Path.GetExtension(fileName).ToLower())
             {
-                return VsItemType.Content;
+                case ".cs":
+                    if (fileName.ToLower().EndsWith(".xaml.cs"))
```

Comment:
I'd suggest using `fileName.EndsWith(".xaml.cs", false, CultureInfo.Invariant)` instead of using ToLower.
https://msdn.microsoft.com/en-us/library/t9h2fbth(v=vs.110).aspx